.github/workflows/e2e-connectivity.yaml

```diff
@@ -0,0 +1,248 @@
+      run: |
+        set -x
+
+        cat >> kind.yaml <<EOF
+        - role: worker
+          image: kindest/node:${{ matrix.k8s-version }}@${{ matrix.kind-image-sha }}
+        - role: worker
+          image: kindest/node:${{ matrix.k8s-version }}@${{ matrix.kind-image-sha }}
+        - role: worker
+          image: kindest/node:${{ matrix.k8s-version }}@${{ matrix.kind-image-sha }}
+        EOF
+    
+    - name: Create KinD Cluster
+      working-directory: ./src/knative.dev/serving
+      run: |
+        set -x
+
+        kind create cluster --config kind.yaml
+
+    - name: Setup NetworkPolicy CNI
+      run: |
+        kubectl apply -f https://docs.projectcalico.org/v3.20/manifests/calico.yaml
+    
+    - name: Setup local registry
+      run: |
+        # Run a registry.
+        docker run -d --restart=always \
+          -p $REGISTRY_PORT:$REGISTRY_PORT --name $REGISTRY_NAME registry:2
+
+        # Connect the registry to the KinD network.
+        docker network connect "kind" $REGISTRY_NAME
+
+        # Make the $REGISTRY_NAME -> 127.0.0.1, to tell `ko` to publish to
+        # local reigstry, even when pushing $REGISTRY_NAME:$REGISTRY_PORT/some/image
+        sudo echo "127.0.0.1 $REGISTRY_NAME" | sudo tee -a /etc/hosts
+    
+    - name: Install Serving & Ingress
+      working-directory: ./src/knative.dev/serving
+      run: |
+        source ./test/e2e-common.sh
+
+        KIND=1
+        INGRESS_CLASS="${{ matrix.kingress }}.ingress.networking.knative.dev"
+        CLUSTER_DOMAIN="${{ matrix.cluster-suffix }}"
+
+        knative_setup
+        test_setup
+
+        echo "INGRESS_CLASS=$INGRESS_CLASS" >> $GITHUB_ENV
+        echo "CLUSTER_DOMAIN=$CLUSTER_DOMAIN" >> $GITHUB_ENV
+        echo "SYSTEM_NAMESPACE=$SYSTEM_NAMESPACE" >> $GITHUB_ENV
+        echo "GATEWAY_OVERRIDE=$GATEWAY_OVERRIDE" >> $GITHUB_ENV
+        echo "GATEWAY_NAMESPACE_OVERRIDE=$GATEWAY_NAMESPACE_OVERRIDE" >> $GITHUB_ENV
+    
+    - name: Apply NetworkPolicy
+      working-directory: ./src/knative.dev/serving
+      run: |
+        set -x
+
+        cat >> netpolicy.yaml <<EOF
+        apiVersion: networking.k8s.io/v1
+        kind: NetworkPolicy
+        metadata:
+          name: default-deny
+          namespace: $SYSTEM_NAMESPACE
+        spec:
+          podSelector:
+            matchExpressions:
+            - key: app
+              operator: In
+              values: [autoscaler, autoscaler-hpa]
+          policyTypes:
+          - Ingress
+          ingress:
+          - from:
+            - namespaceSelector:
+                matchLabels:
+                  # TODO: use this in preference to the projectcalico.org/name label after minVersion >= k8s 1.21.1
+                  # kubernetes.io/metadata.name: $SYSTEM_NAMESPACE
+                  projectcalico.org/name: $SYSTEM_NAMESPACE # calico specific
+        EOF
+        kubectl apply -f netpolicy.yaml
+    
+    - name: Check that autoscaler pods are unreachable
```

Comment:
I might well be missing something, but I don't really understand what this is testing with this in a separate workflow. We're asserting that the NetworkPolicy we added in this workflow applied, but we're not testing that Knative works with that NetworkPolicy enabled, in which case aren't we just testing kubernetes/minikube here? To put it another way, I think the goal is to verify that Knative works even with NetworkPolicy blocking access to the autoscaler, and I don't _think_ this does that any more (tldr I think we might have been on the right track before we moved this to a different file, when we ran the tests with this policy applied).

Reply:
@julz tests seem to work against istio after Kenjiro's changes. I think we would be running a lot of redundant tests by having two separate workflows and the most constricted configuration in this case is the one applying the network policy so it makes sense to me having a single workflow that runs the existing test suite with the network policy applied, I can change it back if @dprotaso has no objections